src/Data/Nat/Logarithm.agda

```diff
@@ -0,0 +1,54 @@
+-----------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Logarithm base 2 and respective properties
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.Nat.Logarithm where
+
+open import Data.Nat 
+open import Data.Nat.Induction using (<-wellFounded)
+open import Data.Nat.Logarithm.Core
+open import Relation.Binary.PropositionalEquality using (_≡_)
+
+------------------------------------------------------------------------
+-- Logarithm base 2
+
+-- Floor version
+
+⌊log₂_⌋ : ℕ → ℕ
+⌊log₂ n ⌋ = ⌊log2⌋ n (<-wellFounded n)
+
+-- Ceil version
+
+⌈log₂_⌉ : ℕ → ℕ
+⌈log₂ n ⌉ = ⌈log2⌉ n (<-wellFounded n)
+
+------------------------------------------------------------------------
+-- Properties of ⌊log₂⌋ and ⌊log₂_⌋
+
+⌊log₂⌊n/2⌋⌋≡⌊log₂n⌋∸1 : ∀ {n} → ⌊log₂ ⌊ n /2⌋ ⌋ ≡ ⌊log₂ n ⌋ ∸ 1
+⌊log₂⌊n/2⌋⌋≡⌊log₂n⌋∸1 {n} = ⌊log2⌋⌊n/2⌋≡⌊log2⌋n∸1 {n}
+
+⌊log₂2*b⌋≡1+⌊log₂b⌋ : ∀ b → b > 0 → ⌊log₂ (2 * b) ⌋ ≡ 1 + ⌊log₂ b ⌋
+⌊log₂2*b⌋≡1+⌊log₂b⌋ b = ⌊log2⌋2*b≡1+⌊log2⌋b b
+
+⌊log₂2^n⌋≡n : ∀ (n : ℕ) → ⌊log₂ (2 ^ n) ⌋ ≡ n
+⌊log₂2^n⌋≡n n = ⌊log2⌋2^n≡n n
+
+------------------------------------------------------------------------
+-- Properties of ⌈log₂⌉ and ⌈log₂_⌉
+
+⌈log₂⌉-mono-≤ : ∀ {a b} → a ≤ b → ⌈log₂ a ⌉ ≤ ⌈log₂ b ⌉
+⌈log₂⌉-mono-≤ p = ⌈log2⌉-mono-≤ p
+
+⌈log₂⌉n/2⌉⌉≡⌈log₂n⌉∸1 : ∀ {n} → ⌈log₂ ⌈ n /2⌉ ⌉ ≡ ⌈log₂ n ⌉ ∸ 1
```

Comment:
The second `⌉` should be a `⌈` and I think `n` should be explicit here?

Reply:
oops good catch! Thank you!

Reply:
Done!